### PR TITLE
Implementing Leaderboard API Integration

### PR DESCRIPTION
**Pull request by ANTOINE1128**
**This pull request adds the necessary functionality to integrate the Leaderboard API into the project, allowing the creation of a new game, retrieving scores, and submitting scores. The implementation follows the provided requirements, utilizing modern JavaScript features such as arrow functions, async/await, and JSON parsing.**

Changes Made:

_Created New Game scores: using the API. The game is given a user-defined name_,

_Implemented "Refresh" Button: Added a "Refresh" button to the user interface, which triggers an API call to retrieve all scores for the game created by the user. The API response is parsed as JSON to extract the relevant data._

_Implemented "Submit" Button: Integrated a form submission button that allows the user to save their score for the game they created. The score is sent to the API for storage._

_Utilized Modern JavaScript Features: Updated the codebase to utilize arrow functions instead of the traditional function keyword. Additionally, async/await keywords are used to handle asynchronous API calls, making the code more concise and readable._

_No Styling Changes: As per the requirements, no styling modifications were made to the project._

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1 +1,40 @@
 import './style.css';
+import addScore from './modules/Addscores.js';
+
+const refreshBtn = document.getElementById('refresh');
+const submitBtn = document.getElementById('submit');
+const insertedName = document.getElementById('inserted-name');
+const insertedScore = document.getElementById('inserted-score');
+const fetchedList = document.getElementById('fetched-list');
+
+// let scoreBoard = [];
```
Comment (DammyShittu) [inline src/index.js]: - [ ] Kindly remove commented code. It is not a good practice as it can make code hard to read.

File: src/index.js [hunk 2]
```diff
@@ -1 +1,40 @@
...
+        list.innerHTML = `
+        <li>${item.user} ${'scored'} ${item.score} ${'points'}</li><br>`;
+        fetchedList.appendChild(list);
+      });
+    });
+};
+
+refreshBtn.addEventListener('click', () => {
+  displayScore();
+});
```
Comment (DammyShittu) [inline src/index.js:30]: - [ ] Your app does not work. When load the app, no scores are displayed (which is okay), but when I click the refresh button nothing happens still. I think you may want to look at your console log errors in the browser.

  image

File: src/index.js [hunk 3]
```diff
@@ -1 +1,40 @@
...
+
+submitBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+  addScore(api, insertedName.value, insertedScore.value);
+  displayScore();
+});
+
```
Comment (DammyShittu) [inline src/index.js:37]: - [ ] The score add feature does not work either. It just refreshes the page when I click submit but when I click the refresh button again, nothing is displayed.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @ANTOINE1128,

## Highlights
✔️ Arrow functions

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/Addscores.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    method: 'POST',
+    body: JSON.stringify({
+      user: name,
+      score,
+    }),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  }).then((response) => response.json());
+};
```
Comment (elisha2kyakpo1) [inline src/modules/Addscores.js]: - [ ] Let's try to understand some requirements for this project.
game

These two requirements mean that we should send a post request (empty), and its return should give us an ID which we should then use as a key. Remember we just want to get only the scores for this game, but your app returns different data which we do not want.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @ANTOINE1128, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

👍 You did well to implement the refresh button and fix the error in the console.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbeeraTahir — APPROVED** [review]
# Status: Approved ✔️✔️✔️  

Hi @ANTOINE1128, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

approved

## Highlights
✔️ Linter checks are passing
✔️ Project is working as expected 👍🙌

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**